axelor-account/src/main/java/com/axelor/apps/account/service/AnalyticDistributionTemplateServiceImpl.java

```diff
@@ -67,6 +76,75 @@ public boolean validateTemplatePercentages(
     return true;
   }
 
+  @Override
+  @Transactional
+  public AnalyticDistributionTemplate personalizeAnalyticDistributionTemplate(
+      AnalyticDistributionTemplate analyticDistributionTemplate, Company company)
```

Comment:
This method is too long and too complex, it will be too hard to maintain. Please create protected methods